src/librustc_codegen_ssa/mir/place.rs

```diff
@@ -250,37 +250,61 @@ impl<'a, 'tcx, V: CodegenObject> PlaceRef<'tcx, V> {
                 ref niche_variants,
                 niche_start,
             } => {
+                // Rebase from niche values to discriminants, and check
+                // whether the result is in range for the niche variants.
                 let niche_llty = bx.cx().immediate_backend_type(encoded_discr.layout);
                 let encoded_discr = encoded_discr.immediate();
-                let (is_niche, niche_discr) = if niche_variants.start() == niche_variants.end() {
-                    // Special case for when we can use a simple equality check,
-                    // which covers null pointers, and needs simpler codegen.
-                    // FIXME(eddyb): check the actual primitive type here.
-                    let encoded_niche = if niche_start == 0 {
-                        // HACK(eddyb): using `c_null` as it works on all types.
+
+                // We first compute the "relative discriminant" (wrt `niche_variants`),
+                // that is, if `n = niche_variants.end() - niche_variants.start()`,
+                // we remap (maybe wrap-around!) `niche_start..=niche_start + n`
```

Comment:
nit: the wording here is slightly awkward, because I read the phrase "wrap-around" as a verb first rather than an adjective, and so  I initially interpret "we remap (maybe wrap-around!) `niche_start..=niche_start + n` as meaning "the remapping we are doing here might wrap-around").

Its role as an adjective becomes clearer after one finishes reading the whole sentence, but I would prefer if you maybe said:

"we remap (the potentially wrapping-around!) `niche_start..=niche_start + n`"
or maybe
"we remap (the potentially overflowing!) `niche_start..=niche_start + n`"

Reply:
"wraparound range" is a specific thing that validity ranges are (e.g. `2..=0` includes everything except `1`).

Maybe this would make more sense if we had a special type for them, but then the `a..=b` syntax wouldn't work...

I think I tried to be terse here to make things fit but I don't think I really have to, so I can reword it a bit.

Reply:
Ah okay I didn't know the adjective use here was a specific term-of-art

Reply:
I wish it was more of a term of art tbh, right now it's kinda niche (pun not intended).